[WIP] Use composer and change directory structure.

SamMousa opened the pull request:
[WIP] Work In Progress, not yet ready for merge!

This PR aims to do 2 things:
1. Remove dependencies from the repository and use composer instead.
2. Change the directory structure so it is easier to secure.

New proposed directory structure:
- root (containing LICENSE, README and composer.json)
-- public (containing entry PHP scripts, all js / image assets that do not need to be protected).
-- protected (containing project code)
-- protected/vendor (containing all dependencies

This allows anyone to simply set the webroot to public and it will work properly and securely.

josephpnc commented:
@nhuhtlb @nhungpham can you quickly review this, if ok then we can merge.

SamMousa commented:
Its not done yet wip --> work in progress
On Apr 6, 2016 7:08 AM, "Joseph Pham Ngoc Cuong" notifications@github.com
wrote:

> @nhuhtlb https://github.com/nhuhtlb @nhungpham
> https://github.com/nhungpham can you quickly review this, if ok then we
> can merge.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/LinxHQ/linxbooks/pull/12#issuecomment-206119640

josephpnc commented:
oops sorry, I misread it. OK, @nhuhtlb @nhungpham @HoDucTien37 please wait until Sam comments this as done.

SamMousa commented:
Okay, I have updated it to work properly.

After you merge, you MUST change the composer.json --> change it so it lists you guys as the author and change the package name, I chose `sam-it/linxbooks` you should probably go for something like `linxhq/linxbooks`.

Side note, I also removed `@session_start()` directive from config since it causes issues with Yii database session table creation.

josephpnc commented:
@HoDucTien37 @nhuhtlb please follow up with what Sam has done and merge. Thanks!
